app/controllers/songs_controller.rb

```diff
@@ -0,0 +1,46 @@
+class SongsController < ApplicationController
+
+  def index
+
+  end
+
+  def new
+
+  end
+
+  def show
+    client = Grooveshark::Client.new({session: session[:groove_session]})
```

Comment:
Since you are accessing the Grooveshark client across multiple controllers, I might create a method in `application_controller.rb` and use that.
```ruby
class ApplicationController
  def groove_client
    @groove_client ||= Groovershark::Client.new(session: session[:groove_session])
  end
end
```
The `||=` allows use to call this method and not worry about whether `.new` was called.  If it wasn't, it'll call it.  If it was, it'll just return the existing one.